src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostClientServiceFactory.RemoteHostClientService.cs

```diff
@@ -187,6 +188,12 @@ private void SetRemoteHostBitness()
                         WellKnownExperimentNames.RoslynOOP64bit);
                 }
 
+                if (!x64)
+                {
+                    // Use 64-bit OOP for systems with 32GiB+ memory
+                    x64 = MemoryHelper.GetMaxPhysicalMemory() >= 32UL * 1024 * 1024 * 1024;
```

Comment:
nit: use shift op?

Reply:
I find this form more readable. Off the top of my head, I'm not sure how many places I need to shift to get 1GiB. 32GiB would be 5 more than that... but still unknown.

Reply:
Hmm, maybe I was confused about the definition of GiB. Isn't one GiB just 2^30? bytes?

Reply:
It is 2<sup>30</sup>, but not everyone knows 30 in the right constant number there.